src/azure-cli/azure/cli/command_modules/vm/_params.py

```diff
@@ -926,6 +926,7 @@ def load_arguments(self, _):
                        help='Space-separated application configuration overrides for each application version ids. '
                        'It should have the same number of items as the application version ids. Null is available for a application '
                        'which does not have a configuration override.')
+            c.argument('treat_deployment_as_failure', nargs='*', help="Space-separated list of true or false corresponding to the application version ids. If set to true, failure to install or update gallery application version operation will fail this operation")
```

Comment:
There must be at least one value passed in for this parameter, right?
```suggestion
            c.argument('treat_deployment_as_failure', nargs='+', help="Space-separated list of true or false corresponding to the application version ids. If set to true, failure to install or update gallery application version operation will fail this operation")
```

Reply:
It is an optional parameter. The only limitation is that if specified, it must have the same number of items as application_version_ids. Calling it with 0 parameters should work technically, because you can call application_version_ids with 0 parameters. But calling it with 0 parameters has the same effect as not calling it. So, it works either way.

For the sake of consistency, because application_configuration_overrides can have 0 parameters, and the behavior is the same with treat_deployment_as_failures. I gave it nargs='*' instead of nargs='+'

Reply:
You mean that using `nargs='*'` is to keep consistent with the definition of `application_version_ids` parameter, right?

Reply:
"You mean that using nargs='*' is to keep consistent with the definition of application_version_ids parameter, right?"

Yes, application_version_ids and also application_configuration_overrides.

Reply:
Got it, thanks